eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionReportWriter.java

```diff
@@ -117,31 +178,81 @@ public void update(ExecutionContext executionContext) throws ItemStreamException
-        for (AccessionWrapper<ISubmittedVariant, String, Long> variant : denormalizedAccessions) {
-            writeVariant(variant.getAccession(), variant.getData());
+        variantsWriter.flush();
+    }
+
+    private void updateChromosomeMappings(List<? extends IVariant> originalVariantsWithReplacedContigs)
+            throws IOException {
+        for (IVariant variantWithContig : originalVariantsWithReplacedContigs) {
+            String originalChromosome = getOriginalChromosome(variantWithContig);
+
+            String previousContig = inputContigsToInsdc.put(originalChromosome, variantWithContig.getChromosome());
+            if (previousContig == null) {
+                // there was no previous entry, so this is not present in the contigs file: write it
+                contigsWriter.write(originalChromosome + "\t" + variantWithContig.getChromosome());
+                contigsWriter.newLine();
+                contigsWriter.flush();
+            } else if (!previousContig.equals(variantWithContig.getChromosome())) {
+                Set<String> contigsAssociatedToTheSameChromosome = duplicatedInputContigsToInsdc.computeIfAbsent(
+                        originalChromosome, key -> new HashSet<>());
+                contigsAssociatedToTheSameChromosome.add(variantWithContig.getChromosome());
+                contigsAssociatedToTheSameChromosome.add(previousContig);
+            }
+
+            String previousChromosome = insdcToInputContigs.put(variantWithContig.getChromosome(), originalChromosome);
+            if (previousChromosome != null && !previousChromosome.equals(originalChromosome)) {
+                Set<String> chromosomesAssociatedToTheSameContig = duplicatedInsdcToInputContigs.computeIfAbsent(
+                        variantWithContig.getChromosome(), key -> new HashSet<>());
+                chromosomesAssociatedToTheSameContig.add(originalChromosome);
+                chromosomesAssociatedToTheSameContig.add(previousChromosome);
+            }
         }
-        fileWriter.flush();
     }
 
-    private void writeVariant(Long id, ISubmittedVariant denormalizedVariant) throws IOException {
-        String vcfLine = variantToVcfLine(id, denormalizedVariant);
-        fileWriter.write(vcfLine);
-        fileWriter.newLine();
+    private String getOriginalChromosome(IVariant variant) {
+        Set<String> originalChromosomes = variant.getSourceEntries()
+                                                 .stream()
+                                                 .map(se -> se.getAttributes().get(
+                                                         ContigToGenbankReplacerProcessor.ORIGINAL_CHROMOSOME))
+                                                 .collect(Collectors.toSet());
+
+        if (originalChromosomes.size() != 1) {
+            throw new IllegalStateException(
+                    "Bug detected: Can not provide the original chromosome of a variant because there should be "
```

Comment:
Suggest to re-phrase to "Bug detected: Multiple original chromosomes were found to be associated with the same variant" and then add variant info, the associated original chromosomes and the replacement used.